auth0/src/main/java/com/auth0/android/authentication/AuthenticationAPIClient.java

```diff
@@ -235,6 +239,43 @@ public AuthenticationRequest login(@NonNull String usernameOrEmail, @NonNull Str
         return loginWithToken(requestParameters);
     }
 
+    /**
+     * Log in a user using the One Time Password code after they have received the 'mfa_required' error.
+     * The MFA token tells the server the username or email, password and realm values sent on the first request.
+     * Example usage:
+     * <pre>
+     * {@code
+     * client.loginWithOTP("{mfa token}", "{one time password}")
+     *      .start(new BaseCallback<Credentials>() {
+     *          {@literal}Override
+     *          public void onSuccess(Credentials payload) { }
+     *
+     *          {@literal}Override
+     *          public void onFailure(AuthenticationException error) { }
+     *      });
+     * }
+     * </pre>
+     *
+     * @param mfaToken the token received in the previous {@link #login(String, String, String)} response.
+     * @param otp      the one time password code provided by the resource owner, typically obtained from an
+     *                 MFA application such as Google Authenticator or Guardian.
+     * @return a request to configure and start that will yield {@link Credentials}
+     */
+    @SuppressWarnings("WeakerAccess")
+    public AuthenticationRequest loginWithOTP(@NonNull String mfaToken, @NonNull String otp) {
```

Comment:
I can't think of a better name right now but it would be good to be consistent on platforms, In swift the counterpart might be defined as `login(withOTP: `

Reply:
Ideally we can have a single "login" method that just sets the `client_id` and the path `/oauth/token`. But then users will need to know for each grant type what parameters are accepted. That's the reason we now have a new login "with otp" method that helps to construct this request.

So in swift what are you doing, is that "withOTP" like a flag?

Reply:
No, so in Swift there are 2 login methods one is /ro the other grant `password-realm`. In this case if I had `login(withOTP otp: String, mfaToken: String)`

The auto-complete would show 3 logins and it's based on the params, in Swift the guideline is to create methods that have meaninful names. You could have `login(otp: String, token: String)` which isn't as informative.